### PR TITLE
Add compiler crash when ignoring the error cause using `_` in an error binding pattern to release note

### PR DESCRIPTION
**Pull request by suleka96**
## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

File: downloads/swan-lake-release-notes/swan-lake-2201.5.0 /RELEASE_NOTE.md [hunk 1]
```diff
@@ -144,6 +144,22 @@ If you have not installed Ballerina, download the [installers](/downloads/#swanl
            sample:main(sample.bal:3)
     ```
 
+- Fixed a bug that resulted in a compiler crash when attempting to ignore the error cause using `_` in an error binding pattern.
```
Comment (praneesha) [inline downloads/swan-lake-release-notes/swan-lake-2201.5.0 /RELEASE_NOTE.md:147]:
```suggestion
- Fixed a bug that resulted in a compiler crash when attempting to ignore the error caused by using `_` in an error binding pattern.
```

Comment (suleka96) [reply downloads/swan-lake-release-notes/swan-lake-2201.5.0 /RELEASE_NOTE.md:147]: error cause is a part of an error: https://ballerina.io/learn/by-example/error-cause/ should we put `error cause` instead if it is misleading?

Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-2201.5.0 /RELEASE_NOTE.md:147]: Shall we add a comma after the word `cause`?

File: downloads/swan-lake-release-notes/swan-lake-2201.5.0 /RELEASE_NOTE.md [hunk 2]
```diff
@@ -144,6 +144,22 @@ If you have not installed Ballerina, download the [installers](/downloads/#swanl
...
+  ```ballerina
+  type SampleErrorData record {|
+    int code;
+    string reason;
+  |};
+
+  type SampleError error<SampleErrorData>;
+
+  public function main() {
+    SampleError sampleErr = error("Transaction Failure", error("Database Error"), code = 20, reason = "deadlock condition");
```
Comment (MaryamZi) [inline downloads/swan-lake-release-notes/swan-lake-2201.5.0 /RELEASE_NOTE.md:158]: Formatting (tab) seems off. Let's also simplify the example, IMO we don't need two detail fields here.

Comment (MaryamZi) [reply downloads/swan-lake-release-notes/swan-lake-2201.5.0 /RELEASE_NOTE.md:158]: May not even need a custom error.

**Pull request closed without merging**